### ISSUE TITLE
Expand auto-saving to more areas

### ISSUE THREAD

**Issue by ricklamers** [orchest/orchest#426]
**Describe the solution you'd like**  
Expand auto-saving to more areas

**What does your solution aim to solve?**  
Currently only the pipeline editor auto-saves the pipeline. Some other edits (like changes to the environment setup script, to name one example) can also be autosaved.

**Comment by yannickperrenet** [comment]
I do want to emphasize here that even though users would no longer have to press "save", they might be so used to having to save changes that they are confused when they don't have to. In case we do choose to autosave we need to make this very clear to the users.

**Comment by yannickperrenet** [comment]
@iannbing The new environments page will also have auto-saving right?

**Comment by iannbing** [comment]
@yannickperrenet Yes, and I'm also thinking about the same thing. Ideally there should be a "saved" indicator somewhere in the UI (gradually building the habit in Orchest users). But currently I'm struggling finding a generic and unintrusive way to inform user.

**Comment by yannickperrenet** [comment]
We can postpone adding a "saved" indicator for now. Would also immediately solve https://github.com/orchest/orchest/issues/178.

Having a "saved" indicator alleviates the need of having a "save" button due to auto-saving, I like it :smile:
